### PR TITLE
Fix bug of possible re-trigger of memory dump

### PR DESCRIPTION
**Pull request by ShellyKa13**
Checking the memory dump file path and seeing it already updated indicates that the request dump already occured or in progress we dont want to simply return we want to skip this duplicate call.

**What this PR does / why we need it**:
This bug was introduced in this commit: https://github.com/kubevirt/kubevirt/pull/8820/commits/f7c1a4c7a2c2a75c184ed3e99f0863b577a4f0ac
of refactoring the virt-launcher to use metadata cache instead of libvirt metadata #8820 
In it when identifying that the metadata is already updated with the dump path instead of skipping the core dump command it just returned silently.
So in case of calling the virt-launcher memory dump before the status of the memory dump updated on the VM as succeeded caused the core dump command to be called again although dump marked as completed on the VM.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes # This should fix flakiness in test:  [sig-storage] VirtualMachineSnapshot Tests [storage-req] With online vm snapshot with memory dump [test_id:8922]should include memory dump in vm snapshot 
The tests sometimes fails because a second memory dump is triggered after already marking to the test that the memory dump completed, and so the snapshot was taken and wasn't succeeded in 30 secs since the VM was paused due to the memory dump and couldn't freeze in order to perform the online snapshot.

**Special notes for your reviewer**:

**Release note**:

```release-note
Fix bug of possible re-trigger of memory dump
```

### CODE REVIEW

File: pkg/virt-launcher/virtwrap/manager.go [hunk 1]
```diff
@@ -1186,6 +1186,10 @@ func (l *LibvirtDomainManager) MemoryDump(vmi *v1.VirtualMachineInstance, dumpPa
 func (l *LibvirtDomainManager) memoryDump(vmi *v1.VirtualMachineInstance, dumpPath string) error {
 	logger := log.Log.Object(vmi)
 
+	skip := l.shouldSkipMemoryDump(dumpPath)
+	if skip {
```
Comment (awels) [inline pkg/virt-launcher/virtwrap/manager.go:1190]: Wouldn't
```go
if l.shouldSkipMemoryDump(dumpPath) {
  return nil
}
```
Work? Not a big deal

Comment (ShellyKa13) [reply pkg/virt-launcher/virtwrap/manager.go]: done

**Comment by akalenyu** [comment]
@ShellyKa13 should we cherrypick this?

**Comment by ShellyKa13** [comment]
yep
/cherrypick release-0.59

**Pull request merged**